lib/messages/commands/tx.js

```diff
@@ -2,28 +2,30 @@
 
 var Message = require('../message');
 var inherits = require('util').inherits;
+var bitcore = require('bitcore');
+var $ = bitcore.util.preconditions;
+var _ = bitcore.deps._;
 
 /**
- * @param {Object|Transaction=} options - If is an instance of Transaction will use as options.transaction
- * @param {Transaction=} options.transaction - An instance of a Transaction
+ * @param {Transaction=} arg - An instance of Transaction
+ * @param {Object} options
  * @param {Number} options.magicNumber
  * @extends Message
  * @constructor
  */
-function TransactionMessage(options) {
-  Message.call(this, options);
+function TransactionMessage(arg, options) {
+  Message.call(this, arg, options);
   this.command = 'tx';
   this.magicNumber = options.magicNumber;
   this.Transaction = options.Transaction;
-
-  var transaction;
-  if(options instanceof this.Transaction) {
-    transaction = options;
-  } else {
-    transaction = options.transaction;
+  $.checkArgument(
+    _.isUndefined(arg) || arg instanceof this.Transaction,
+    'An instance of Transaction or undefined is expected'
```

Comment:
It's necessary to be able to construct a message without the `arg` set.

There is also another test here that constructs messages without the arg: https://github.com/bitpay/bitcore-p2p/blob/master/test/messages/index.js#L89

More importantly though a message needs to be able to be constructed partially the options only, and then modified with a payload using those options: https://github.com/braydonf/bitcore-p2p/blob/bug/transactions-message/lib/messages/builder.js#L77